packages/test/test-end-to-end-tests/src/test/attachLifecycle.spec.ts

```diff
@@ -0,0 +1,189 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    createLoader,
+    ITestFluidObject,
+    timeoutPromise,
+} from "@fluidframework/test-utils";
+
+import {generatePairwiseOptions} from "@fluidframework/test-pairwise-generator";
+import { describeFullCompat } from "@fluidframework/test-version-utils";
+import { IResolvedUrl } from "@fluidframework/driver-definitions";
+import { ISharedMap, IValueChanged, SharedMap } from "@fluidframework/map";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { AttachState } from "@fluidframework/container-definitions";
+
+// during these point succeeding objects won't even exist locally
+const ContainerCreated = 0
+const DatastoreCreated = 1;
+const MapCreated = 2
+
+//these points are after all objects at least exist locally
+const sharedPoints = [3, 4, 5];
+
+const testConfigs =
+    generatePairwiseOptions({
+        containerAttachPoint:[ContainerCreated, DatastoreCreated, ... sharedPoints],
+        containerSaveAfterAttach: [true, false],
+        datastoreAttachPoint: [DatastoreCreated, ... sharedPoints],
+        datastoreSaveAfterAttach: [true, false],
+        mapAttachPoint: [MapCreated, ... sharedPoints],
+        mapSaveAfterAttach: [true, false],
+    });
+
+describeFullCompat("Validate Attach lifecycle", (getTestObjectProvider) => {
+    before(function (){
+        const provider  = getTestObjectProvider();
+        switch(provider.driver.type){
+            case "local":
+                break;
+            default:
+                this.skip();
```

Comment:
hanging forever on tinylicious, i also worry about it DOSing odsp. will file a bug once its in

Reply:
#9133